app/components/quiz/QuizQuestion.js

```diff
@@ -19,20 +19,24 @@ export default React.createClass({
     return {colors: randomColors, isSelectedFlags: isSelected};
   },
 
-  answerClicked: function(id, selected) {
+  answerProcessed: function(id, selected, text=null) {
```

Comment:
hm. maybe break this function down into different functions based on type, for readability?